woocommerce/woocommerce-ios
# Coupon Editing: set optional values in the coupon model under the storage layer

Fixes #6749 

### Description
While I was working on the fields validation in the Edit Coupon > Usage Restrictions screen, I discovered that some fields like `usageLimit` and `limitUsageToXITems` are always set to 0 also if on the web the values are not populated and the API response contains null values for these fields.
So, looking at the Coupon model in Core Data, these fields are declared mandatory and we set 0 as the default value.
In this PR I created a new storage model (n° 68), changing these properties declaring them as optional, and setting `null` as the default value. The property declared as optional are: `usageLimit`, `usageLimitPerUser` and `limitUsageToXItems`.


### Testing instructions
1. Open a coupon detail, where one or more of the fields above are `null`.
2. Open the edit screen, then the Usage Restrictions screen.
3. The fields mentioned should be empty (with a placeholder displayed).

### Screenshots
<img src="https://user-images.githubusercontent.com/495617/166916822-dcaa010a-32ad-4db8-ace3-938d01b76538.png" width=300 />




---
